pkg/components/exec/file.go

```diff
@@ -75,24 +76,32 @@ func FindExecELF(p *services.ProcessInfo, svcID svc.Attrs, k8sEnabled bool) (*Fi
 		return nil, err
 	}
 
-	file.Service.EnvVars = envVars
-	if svcName, ok := file.Service.EnvVars[envServiceName]; ok {
-		// If Kubernetes is enabled we use the K8S metadata as the source of truth
-		if !k8sEnabled {
-			file.Service.UID.Name = svcName
-		}
+	setServiceEnvVariables(&file.Service, envVars, k8sEnabled)
+
+	return &file, nil
+}
+
+func setServiceEnvVariables(service *svc.Attrs, envVars map[string]string, k8sEnabled bool) {
```

Comment:
Can we use return values instead of having the function perform a side-effect on the parameter?


```suggestion
func setServiceEnvVariables(service *svc.Attrs, envVars map[string]string, k8sEnabled bool) *svc.Attrs {
```

This way the function signature clearly communicates what it accepts and what value it will produce.